src/utils.js

```diff
@@ -1,13 +1,48 @@
+// Create a local reference to common methods we'll want to use later.
+export const slice = Array.prototype.slice;
+export const hasOwn = Object.prototype.hasOwnProperty;
+
+// Splices `insert` into `array` at index `at`.
+export function splice(array, insert, at) {
+  at = Math.min(Math.max(at, 0), array.length);
+  const tail = Array(array.length - at);
+  const length = insert.length;
+  let i;
+  for (i = 0; i < tail.length; i++) tail[i] = array[i + at];
+  for (i = 0; i < length; i++) array[i + at] = insert[i];
+  for (i = 0; i < tail.length; i++) array[i + length + at] = tail[i];
+}
```

Comment:
According to https://github.com/jashkenas/backbone/pull/4027#issuecomment-217508725, `splice` is sloooow.

Reply:
Ah I misread the comment - `splice` in a loop is slow, which is 100% accurate because then you're repeatedly copying the tail. Much better to bailout to a manual splice insertion implementation when there are too many items.